### PR TITLE
fix: Check revenant ability costs correctly

### PR DESCRIPTION
**Pull request by DisposableCrewmember42**
## About the PR
This PR fixes the Revenant ability check to actually cancel an action when the Revenant does not have enough essence.

## Why / Balance
The status quo seems unintentional; the Revenant can currently just spam abilities, even if they have no Essence. This obviously nerfs the Revenant, requiring them to actually steal essences in order to be able to destroy the station in a timely manner.

## Technical details
Currently, the negative price value is checked against, such that Revenants can always use their abilities. This PR makes the default prices non-negative, such that the check works, and simply multiplies the prices by -1 before passing to ChangeEssenceAmount.

## Media

https://github.com/user-attachments/assets/8b22f2d6-b9c3-480a-b5b7-921e0b24960e

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

- The RevenantComponent is changed such that the ability cost attributes, defileCost, overloadCost, blightCost and malfunctionCost have to be positive in order to work properly. If your server changes these values on the Revenant prototype, they will have to be adjusted.

**Changelog**

:cl: DiposableCrewmember42
- fix: Fixed Revenant ability cost checks. Revenants can no longer spam abilities without Essence.

### CODE REVIEW

File: Content.Server/Revenant/EntitySystems/RevenantSystem.cs [hunk 1]
```diff
@@ -170,7 +170,7 @@ private bool TryUseAbility(EntityUid uid, RevenantComponent component, FixedPoin
             }
         }
 
-        ChangeEssenceAmount(uid, abilityCost, component, false);
+        ChangeEssenceAmount(uid, abilityCost * -1, component, false);
```
Comment (EmoGarbage404) [inline Content.Server/Revenant/EntitySystems/RevenantSystem.cs:173]:
```suggestion
        ChangeEssenceAmount(uid, -abilityCost, component, false);
```

Comment (DisposableCrewmember42) [reply Content.Server/Revenant/EntitySystems/RevenantSystem.cs]: :+1:

**Comment by muburu** [comment]
revenant is already absolutely gutted as an antag please stop

**Comment by DogZeroX** [comment]
Yeah please stop fixing the antag, keep the bugs.

**Pull request merged**